cmd/csi-provisioner/csi-provisioner.go

```diff
@@ -59,12 +59,32 @@ var (
 	workerThreads        = flag.Uint("worker-threads", 100, "Number of provisioner worker threads, in other words nr. of simultaneous CSI calls.")
 	operationTimeout     = flag.Duration("timeout", 10*time.Second, "Timeout for waiting for creation or deletion of a volume")
 	provisioner          = flag.String("provisioner", "", "This option is deprecated")
+	inTreePluginName     = flag.String("supersedes-in-tree-plugin-name", "", "An in-tree plugin that is superseded by this CSI plugin")
 
 	featureGates        map[string]bool
 	provisionController *controller.ProvisionController
 	version             = "unknown"
 )
 
+// check that the CSI plugin handles migration for the inTreePluginName according to:
+// https://github.com/kubernetes/csi-translation-lib/blob/master/translate.go#L28
+// A mismatch will lead to failed migration API calls like TranslateCSIPVToInTree. So warn of any mismatch but don't exit.
+func checkCSITranslationLibEntries(csiPluginName, inTreePluginName string) {
+	if !csitranslationlib.IsMigratedCSIDriverByName(csiPluginName) {
+		klog.Warningf("Provisioner name: %s not registered in CSI translation library for performing migrations for %s", csiPluginName, inTreePluginName)
+		return
+	}
+	inTreePluginNameFromLib, err := csitranslationlib.GetInTreeNameFromCSIName(csiPluginName)
```

Comment:
If we can query in-tree plugin name from CSI driver name, why do we need `--supersedes-in-tree-plugin-name`  options? Should be `--feature-gate` enough? It allows us to enable it by default during beta.

Reply:
oops this is complete opposite of my comment from earlier. I didn't think about enabling it by default during beta. Sorry @ddebroy could you change it back... my bad

Reply:
Will revert back to querying behavior. That will ensure a single source of truth i.e. the translation library and enablement of migration by default. As far as feature gates is concerned, this will depend on the in-tree feature gates to be enabled for individual plugins (besides migration in general) for the PV controller to pass PVCs with in-tree provisioners to external provisioners.